src/datastore/src/main/java/org/locationtech/geogig/geotools/data/reader/SimplifyingGeometryReplacer.java

```diff
@@ -0,0 +1,88 @@
+package org.locationtech.geogig.geotools.data.reader;
+
+import com.google.common.base.Function;
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Envelope;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.geom.MultiPoint;
+import com.vividsolutions.jts.geom.MultiPolygon;
+import com.vividsolutions.jts.geom.Point;
+import com.vividsolutions.jts.geom.Polygon;
+import com.vividsolutions.jts.simplify.DouglasPeuckerSimplifier;
+import org.opengis.feature.simple.SimpleFeature;
+
+public class SimplifyingGeometryReplacer implements Function<SimpleFeature, SimpleFeature> {
+
+        double distance;
+
+        GeometryFactory geometryFactory;
+
+        public SimplifyingGeometryReplacer(double distance, GeometryFactory geometryFactory) {
+                this.distance = distance;
+                this.geometryFactory = geometryFactory;
+        }
+
+        @Override public SimpleFeature apply(SimpleFeature input) {
+                Geometry g = (Geometry) input.getDefaultGeometry();
+                Geometry newGeometry;
+
+                if ((g != null) || (g instanceof Point) || (g instanceof MultiPoint) || (!g
```

Comment:
I _think_ you intended to negate these two?: `|| (g instanceof Point) || (g instanceof MultiPoint)`, like in, looks like all geoms are being caught due to the `!g.isEmpty()` but you wanted to bypass points?
If so, what about `if( g != null && !g.isEmpty() && g.getDimension() > 0){...`

Reply:
ya - that looks like its problematic